src/Core/Framework/Webhook/Hookable/HookableEntityInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Webhook\Hookable;
+
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * Interface that EntityDefinitions can implement to declare themselves as hookable.
+ * This allows for dynamic discovery of hookable entities without maintaining a hardcoded list.
+ *
+ * @internal only for use by the app-system
+ */
+#[Package('framework')]
+interface HookableEntityInterface
+{
+    /**
+     * Returns whether this entity should be hookable for webhook events.
+     * This method allows entities to conditionally enable/disable their hookable status.
+     */
+    public function isHookable(): bool;
```

Comment:
This method is unnecessary if the Definition already implements the interface tbh, or the interface won't be needed if the isHookable method is defined (default as false) in EntityDefinition parent class

Reply:
Another approach could be to create a new tag `shopware.entity.hookable` for hookable entities definitions, so you don't need to touch existing Definition classes. The downside is you can't have additional method like `getHookableEventPrivileges`